Use PricePlan component for displaying prices and fix various CSS issues

delawski opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to to-test.md in a new commit as part of your PR. -->
This PR supersedes #14085 

The main purpose of this PR is to fix product price rendering. It adds a simplified version of `PlanPrice` component from Calypso and makes use of it in the `SingleProductBackup` component.

Also, the `SingleProductBackup` component has been cleaned up in a few places. The CSS/appearance of the product card has been improved so that now it matches the designs in Figma more closely.

**Before**
![Screenshot 2019-11-21 at 15 24 53](https://user-images.githubusercontent.com/478735/69346465-1ae1cf00-0c73-11ea-8e4d-42c2640908ef.png)

**After**
![Screenshot 2019-11-21 at 15 14 59](https://user-images.githubusercontent.com/478735/69346422-01d91e00-0c73-11ea-9629-fbc5e59d1856.png)


Fixes n/a

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->
* Migrate `PlanPrice` UI component from Calypso
* Use `PricePlan` component for displaying prices in `SingleProductBackup`
* Clean up `SingleProductBackup`
* Fix various CSS issues in `SingleProductBackup`

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here.

#### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Go to /wp-admin/admin.php?page=jetpack#/plans
* Confirm that the prices/price ranges in Jetpack Backup product card are displayed correctly
* Confirm that the Jetpack Backup product card looks as in the designs in Figma

#### Proposed changelog entry for your changes:
<!-- Please do not leave this empty. If no changelog entry needed, state as such. -->
* Introduce `PlanPrice` UI component.
* Improve `SingleProductBackup` appearance

delawski committed:
Migrate PlanPrice UI component from Calypso

delawski committed:
Use PricePlan component for displaying prices. Fix various CSS issues

enejb commented:
Great work Piotr! 

In my testing this worked really well. 
The mobile case looks a bit funny. But before we fix it lets ask what the expected behavior is here. 

Since the current plans page just leads to the calypso plans page on mobile. 

This is what I see
![Screen Shot 2019-11-21 at 3 57 44 PM](https://user-images.githubusercontent.com/115071/69349425-018f5180-0c78-11ea-926a-c8bc157932b7.png)

delawski commented:
@enejb That's a clear bug. I'm going to fix those minor issues similarly to what we have in Calypso 👍

jeherve commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -61,89 +76,65 @@ export function SingleProductBackup( { products, upgradeLinks } ) {
+			{ isDiscounted && (
+				<PlanPrice currencyCode={ currencyCode } rawPrice={ discountedPrice } discounted />
+			) }
+			<div className="single-product-backup__price-group-billing-timeframe">
```
I don't know if that's ideal. In some languages "per year" can come before the amount.

delawski replied:
My reasoning here was that if there were no issues in Calypso with this approach, we should be good to re-use it here.

jeherve replied:
If we follow the same structure in Calypso, we may want to fix it there as well, to improve localization. But I'll let you all decide.

delawski replied:
I'm going to change it to allow arbitrary order of price and billing time frame.

enejb replied:
Thanks 👍

robertf4 commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -12,25 +11,37 @@ import { get } from 'lodash';
+
+	const priceDaily = get( products, [ 'jetpack_backup_daily', 'cost' ], '' );
+	const priceDailyMonthly = get( products, [ 'jetpack_backup_daily_monthly', 'cost' ], '' );
+	const priceDailyMonthlyPerYear = priceDailyMonthly * 12;
```
In the case where priceDaily doesn't exist it becomes `''`, so `priceDailyMonthly` should probably also be `''` in this case instead of `0`.

```suggestion
	const priceDailyMonthlyPerYear = '' === priceDailyMonthly ? '' : priceDailyMonthly * 12;
```

robertf4 commented on _inc/client/plans/single-product-backup.jsx:
```diff
@@ -12,25 +11,37 @@ import { get } from 'lodash';
+	const priceDailyMonthlyPerYear = priceDailyMonthly * 12;
+	const priceRealtime = get( products, [ 'jetpack_backup_realtime', 'cost' ], '' );
+	const priceRealtimeMonthly = get( products, [ 'jetpack_backup_realtime_monthly', 'cost' ], '' );
+	const priceRealtimeMonthlyPerYear = priceRealtimeMonthly * 12;
```
```suggestion
	const priceRealtimeMonthlyPerYear = '' === priceRealtimeMonthly ? '' : priceRealtimeMonthly * 12;
```

delawski committed:
Address code review feedback and improve mobile layout of the card.

delawski commented on _inc/client/components/plans/plan-price/index.jsx:
```diff
@@ -0,0 +1,85 @@
+
+		// "Normalize" the input price or price range.
+		const rawPriceRange = Array.isArray( rawPrice ) ? rawPrice.slice( 0, 2 ) : [ rawPrice ];
+		if ( rawPriceRange.includes( 0 ) ) {
```
Good catch! I've ported this code from Calypso but haven't considered some possibilities. Let's take care this in a subsequent PR or when we run into an issue with a `$0` price so that this PR is not blocked.

delawski committed:
Update _inc/client/plans/single-product-backup.jsx

Co-Authored-By: robertf4 <42627630+robertf4@users.noreply.github.com>

delawski committed:
Update _inc/client/plans/single-product-backup.jsx

Co-Authored-By: robertf4 <42627630+robertf4@users.noreply.github.com>

delawski commented:
@jeherve I've addressed the feedback about the i18n of the price and the billing interval. Now it's possible to shuffle the order of those two pieces depending on a given language needs. [Please have a look into the code](https://github.com/Automattic/jetpack/pull/14089/files#diff-7457a6cdabf24fa36923c20d4c9b101cR98-R109) to see how it looks now.

@enejb I've also improved the mobile layout of the card. Now it behaves more-less like the card in Calypso. I've also made use of the `.dops-card` class name to make the card look exactly the same as other cards in Jetpack.

Here are two screenshots of an updated card.

**Mobile** 
![Screenshot 2019-11-21 at 20 40 29](https://user-images.githubusercontent.com/478735/69371317-ea645a00-0c9f-11ea-869c-6071815557d6.png)

**Desktop**
![Screenshot 2019-11-21 at 20 40 39](https://user-images.githubusercontent.com/478735/69371327-ed5f4a80-0c9f-11ea-92f0-befe2eaf8d08.png)

@robertf4 Thank you for your review. I've responded to your comments and committed your changes 👍

enejb commented:
Great Work Piotr! 
I tested this and it works as expected!

enejb reviewed (approved):
Great work here. I tested this and it worked as expected for me. Lets ship this change and iterate if we find something that wasn't quite right.

robertf4 commented:
> Great work here. I tested this and it worked as expected for me. Lets ship this change and iterate if we find something that wasn't quite right.

I like this, it will keep us moving forward.

crunnells commented:
Design looks good!

simison commented:
>It adds a simplified version of `PlanPrice` component from Calypso

Did you consider moving the component to [`@automattic/components`](https://github.com/Automattic/wp-calypso/tree/5eb5df59d970e69d94726bebecfde46b67cc1e1c/packages/components) in Calypso and re-using it from there?

robertf4 commented:
> Did you consider moving the component to [`@automattic/components`](https://github.com/Automattic/wp-calypso/tree/5eb5df59d970e69d94726bebecfde46b67cc1e1c/packages/components) in Calypso and re-using it from there?

That is a fantastic idea and I think it should be done, but with code freeze being Monday we can't afford to do it this release.
